storage/google/cloud/storage/_helpers.py

```diff
@@ -33,7 +34,12 @@ class _PropertyMixin(object):
     """
 
     def __init__(self, name=None):
-        self.name = name
+        if name is None or (re.match(r'\w', name[0]) and
```

Comment:
The rules that are worth checking are probably these (being looser than the API is on the last one):

> Bucket names must contain only lowercase letters, numbers, dashes (-), underscores (_), and dots (.). Names containing dots require verification.
> Bucket names must start and end with a number or letter.
> Bucket names must contain 3 to 63 characters. Names containing dots can contain up to 222 characters, but each dot-separated component can be no longer than 63 characters.

A regex that will get close enough on this (e.g. catch common issues but not fail on every single invalid name) would be:

```python
re.match(r'^[a-zA-Z0-9][a-zA-Z0-9_.-]{1,220}[a-zA-Z0-9]$', name)
```